source/material_model/steinberger.cc

```diff
@@ -675,14 +368,31 @@ namespace aspect
           lateral_viscosity_file_name  = prm.get ("Lateral viscosity file name");
           use_lateral_average_temperature = prm.get_bool ("Use lateral average temperature for viscosity");
           n_lateral_slices = prm.get_integer("Number lateral average bands");
-          interpolation        = prm.get_bool ("Bilinear interpolation");
           latent_heat          = prm.get_bool ("Latent heat");
           reference_eta        = prm.get_double ("Reference viscosity");
           min_eta              = prm.get_double ("Minimum viscosity");
           max_eta              = prm.get_double ("Maximum viscosity");
           max_lateral_eta_variation    = prm.get_double ("Maximum lateral viscosity variation");
           thermal_conductivity_value = prm.get_double ("Thermal conductivity");
 
+
+          // Parse the table lookup parameters
+
+          // The Steinberger material model currently assumes that all the
+          // compositional fields correspond to materials with
+          // PerpleX lookup tables.
+          // Therefore the first composition index is hard-coded as zero and
+          // a background field exists if the number of files is one greater
+          // then the number of compositional fields.
+          prm.set("Material file format", "perplex");
+          prm.set("Index of first mass fraction compositional field", "0");
+          if (material_file_names.size() == this->n_compositional_fields() + 1)
+            prm.set("Background material", "true");
+          else
+            prm.set("Background material", "false");
```

Comment:
Is there a reason you set this here rather than in the equation of state? 
With these prm.set functions, what happens if a user sees that option in the parameter viewer, and decides they want to use it? Is this then overwritten here? That's probably not great for debugging. Wouldn't it be better to throw an Assert instead if someone tries to use this material model with the HeFesTo reader? The default is perplex anyway.

Reply:
Oh, good point. Steinberger should be able to use either Perple_X/HeFESTo tables, so I'll remove that prm.set(). The others I'll change to AssertThrows.